jupyterlab_extension/services.py

```diff
@@ -10,6 +10,42 @@
 PROJECTS_ENDPOINT = os.getenv("PROJECTS_ENDPOINT", "http://projects.platiagro:8080")
 
 
+def find_task_by_name(os_path):
+    """
+    Get the task name from the notebook using the PlatIAgro Projects API.
+
+    Parameters
+    ----------
+    os_path : str
+
+    Returns
+    -------
+    dict
+        The task_id.
+
+    Raises
+    ------
+    ConnectionError
+        When the request did not succeed.
+    HTTPError
+        When the request did not succeed.
+    """
+    # only do this for Experiment.ipynb
+    match = re.search(r"/tasks/(.*?)/Experiment.ipynb", os_path)
+
+    if match:
+        task_name = match.group(1)
+
+        params = {'name': task_name}
+        r = requests.get(f"{PROJECTS_ENDPOINT}/tasks", params=params)
+        r.raise_for_status()
+        data = r.json()
+        if len(data["tasks"]) > 0:
+            data_tasks = data["tasks"][0]
+
+            return data_tasks.get("uuid")
```

Comment:
@dnlcesilva você pode fazer +1 teste/ajuste por favor?

Caso existam tarefas com nomes parecidos, por exemplo, `"Tarefa"` e `"Tarefa - Cópia"` 
se você editar a `"Tarefa"`, o retorno do `requests.get` provavelmente vai trazer 2 itens (já que ele procura por substring).
Pode ser que o índice 0 seja a `"Tarefa - Cópia"` .

Você pode, por favor, adicionar um `for task in tasks:` e um `if task["name"] == task_name` pra evitar este erro?

Reply:
@fberanizo alterado!